sdk/identity/azure-identity/inc/azure/identity/workload_identity_credential.hpp

```diff
@@ -61,42 +80,29 @@ namespace Azure { namespace Identity {
     std::string m_tokenFilePath;
 
     explicit WorkloadIdentityCredential(
-        std::string tenantId,
-        std::string const& clientId,
-        std::string const& tokenFilePath,
-        std::string const& authorityHost,
+        std::string const& tenantIdOptions,
+        std::string const& clientIdOptions,
+        std::string const& authorityHostOptions,
```

Comment:
There are a LOT of parameters to this constructor, and it's EXTREMELY easy to confuse them (tenantId and clientId are both GUIDs for example). They might be better expressed as an `Options` structure.

Reply:
Why do we need the private "workhorse" ctor? Why can't we just initialize the appropriate fields in the public ctor?

Reply:
We don't **need it** per say. I imagine it's to avoid code duplication for the ctors. I have been following the same coding style as the existing credential types. I don't have a strong opinion one way or another atm, and am OK with removing this private ctor and simply inline the implementation, like you suggested. It keeps things simple.

Reply:
Looking at the other identity packages, none of the delegating constructors have the 'xxxOptions' convention. The naming of the parameters should be consistent with all the other identity classes.

Having said that, there is a LOT of opportunity for simplification across all the identity constructors, which I believe can be done with a non-breaking set of changes. 

Let's take ClientSecretCredential for example.

The two ctors are:

- `ClientSecretCredential(tenantId,clientId,clientSecret,TokenCredentialOptions={})`
- `ClientSecretCredential(tenantId,clientId,clientSecret,ClientSecretCredentialOptions)`

These then forward to a delegating constructor which either passes defaulted values from the ClientSecretCredentialOptions and the TokenCredentialOptions or the values from ClientSecretCredentialOptions. 

Since `ClientSecretCredentialOptions` derives from `TokenCredentialOptions` these could be combined into:

- `ClientSecretCredential(tenantId,clientId,clientSecret,ClientSecretCredentialOptions={})

The current delegating constructor for `ClientSecretCredential` does the following:

- Initialize m_clientCredentialCore with the tenantId, authorityHost (from ClientSecretCredentialOptions), and additionallyAllowedTenants (also from CLientSecretCredentialOptions).
- Initializes m_tokenCredentialImpl from options (which could be a CLientSecretCredentialOptions since it derives from TokenCredentialOptions).
- Initializes m_requestBody from the clientId and clientSecret (both parameters).

There's no reason that the constructor needs the values in ClientSecretCredentialOptions to be pulled out as separate parameters.

Anyway, it's just a thought about how to simplify both this and the other credential types.